src/js/modules/CartPopup.js

```diff
@@ -59,90 +61,94 @@ class CartPopup {
     return totalPrice;
   }
 
-  increaseCartItemQuantity(item, button, index) {
-    const self = this;
+  increaseCartItemQuantity(products, item, button, index) {
+    console.log("button", button)
+    const self= this;
     if (item.cartQuantity < item.quantity) {
       button.classList.remove("disable");
       item.cartQuantity += 1;
       console.log("item cart quantity", item.cartQuantity);
-      this.currentCartProduct[index] = item;
-      updateCartProduct(this.currentCartProduct);
-      self.loadCartProducts()
+      products[index] = item;
+      updateCartProduct(products);
     } else {
       button.classList.add("disable");
+      self.loadCartProducts();
     }
+    this.loadCartProducts();
   }
-  decreaseCartItemQuantity(item, button, index) {
+  decreaseCartItemQuantity(products, item, button, index) {
     const self = this;
     if (item.cartQuantity > 0) {
       button.classList.remove("disable");
       item.cartQuantity -= 1;
       console.log("item cart quantity", item.cartQuantity);
 
-      this.currentCartProduct[index] = item;
-      updateCartProduct(this.currentCartProduct);
-      self.loadCartProducts();
+      products[index] = item;
+      updateCartProduct(products);
     } else {
-      button.classList.add("disable");
+      self.deleteCart(products, item);
+
     }
+    this.loadCartProducts();
   }
-  deleteCart(item, index) {
-    console.log("before delete", this.currentCartProduct);
-    console.log('target', item);
-    console.log("item in delete", item, "index", index);
-    this.currentCartProduct = this.currentCartProduct.filter(item => item.id !== index);
-    console.log('after', this.currentCartProduct)
-    updateCartProduct(this.currentCartProduct);
+  deleteCart(products, item) {
+    console.log("before delete", products);
+    products = products.filter(
+      (product) => product !== item
+    );
+    console.log("after delete", products);
+    updateCartProduct(products);
     this.loadCartProducts();
   }
-  async initIncreaseCartEvent() {
+  async initIncreaseCartEvent(products) {
     let buttons = document.querySelectorAll(".js-cart-increase");
```

Comment:
Only init click event and get product id then using according to function, remove async if don't need to use await here